### ISSUE TITLE
Found a small bug

### ISSUE THREAD

**Issue by mikhoul** [ddawson/Saved-Password-Editor#26]
I was looking in the FF console for something else and I see this warning:  http://i.imgur.com/xF6j08k.png

`NS_ERROR_ILLEGAL_VALUE: Component returned failure code: 0x80070057 (NS_ERROR_ILLEGAL_VALUE) [nsISyncMessageSender.sendSyncMessage]`

and the culprit: http://i.imgur.com/UlAYIAY.png

Best Regards :smile:

**Comment by ddawson** [comment]
Hmm, is that the problem? I admit, I didn't research the matter too deeply, since I needed to get e10s working quickly. At the time, the documentation about this said it's okay to send synchronous messages *from frame scripts*, as long as the user won't notice any slowdown in the interface. Also, looking in the Add-on SDK code, I found it does just what I'm doing there, so I figured that's a reasonable way to do it.

Well, either they changed things so context menus are handled without `sendSyncMessage`, or something else went wrong. What version of Firefox/SeaMonkey did you see this on?

Alas, I won't be able to do much for at least a few days. Hardware failure has rendered my PC inoperative. I have to wait until I get a replacement part.

**Comment by mikhoul** [comment]
It was on FF 48.0.2.

I did not see any impact, I was just debugging another script with the console and I see this "strange" error so I just tell to myself "I will tell it to the dev".

It work fine for me and if you are working on E10 version maybe it doesn't worth the time to "fix" it since it's will be "scrapped" soon !  :octocat: 

Have a Great Day ! :smile:

**Comment by mikhoul** [comment]
@ddawson   Hi Dan, 

I have more information since I was troubleshooting my "Firefox setup" since I had trouble with the context menu your addon was not the main culprit but later I found that sometime with other addon it make the context menu hang for 25-30 sec with high CPU.

So today I took a closer look to find what was happening , I use "Saved Password Editor" since many years but I use it mainly to EDIT passwords from the tools menu, I was even not aware that a context menu function was inside before today :smiley_cat: 
That's why last week I told you that even with the error everything was working fine because I never used the context menu. 

So I took a look at the error console and I see that I should see a context menu to select the password or the username from the fields in Login page.

But I don't see any Item in the "context menu". 

I use Win7/64 with FF49.0.1 

Here's a short screencast to show you what happen exactly: http://screencast-o-matic.com/watch/cD6XFAj4qg


Best Regards ! :octocat:

**Comment by mikhoul** [comment]
Opps I forgot to ask you a question,

Do you know in meantime if I can disable the context menu for SPE safely by modding the frame-scrips.js   :question: 


```
"use strict";

const Ci = Components.interfaces, Cu = Components.utils;
Cu.import("resource://gre/modules/Services.jsm");
Cu.import("resource://savedpasswordeditor/SavedPasswordEditor-frame.jsm");

Services.obs.addObserver(
  function (aSubject) {
    if (aSubject.wrappedJSObject) aSubject = aSubject.wrappedJSObject;
    var target = aSubject.event.target;
    if (Ci.nsIDOMXULElement
        && target instanceof Ci.nsIDOMXULElement)  // SeaMonkey, why?
      target = target.triggerNode;

    sendSyncMessage(
      "SavedPasswordEditor:contextshowing",
      SavedPasswordEditor.getFormData(target));
  },
  "content-contextmenu",
  false);

addMessageListener(
  "SavedPasswordEditor:scanforloginforms",
  SavedPasswordEditor.scanForLoginForms);
```

Let me know if there is some hack I can use till you come with a fixed version for the context menu. :smile:

**Comment by ddawson** [comment]
Okay, I've written a fix for this. But with extension signing now being unconditionally enforced in release and beta versions, you will not be able to modify the add-on yourself. It's possible to build Firefox with the code to check the `xpinstall.signatures.required` preference enabled; there are also [special unbranded builds](https://wiki.mozilla.org/Add-ons/Extension_Signing#Unbranded_Builds) available configured that way for add-on development. But either way, you would have to go to the trouble of installing it, and you would open yourself to the sorts of third-party add-on shenanigans that this enforcement was intended to prevent.

Anyway, in case you find it useful, you can see the fix in commit e84cea0.

**Comment by mikhoul** [comment]
@ddawson   Thanks a lot for the fix ! :+1: 

I have no problem with "extension signing" since I like to "tweak" some addons my self when the code is simple I use this **[trick ](http://www.ghacks.net/2016/08/14/override-firefox-add-on-signing-requirement/)**to bypass it since it allow me to use Firefox with my locale language **AND** I can use the release channel and everything is updated as usual :smiley_cat: 

I will package the XPI file since the the fix.

Have a nice day ! :sun_behind_small_cloud:
